app/src/androidTest/java/cotel/comonadic_ui/TodosListFragmentTest.kt

```diff
@@ -0,0 +1,123 @@
+package cotel.comonadic_ui
+
+import android.support.test.rule.ActivityTestRule
+import android.support.test.runner.AndroidJUnit4
+import android.view.View
+import com.agoda.kakao.KButton
+import com.agoda.kakao.KRecyclerItem
+import com.agoda.kakao.KRecyclerView
+import com.agoda.kakao.KTextView
+import com.agoda.kakao.Screen
+import org.hamcrest.Matcher
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class TodosListFragmentTest {
+
+  @Rule
+  @JvmField
+  val rule = ActivityTestRule(MainActivity::class.java, true, false)
+
+  val todosListScreen = TodosListScreen()
+
+
+  @Test
+  fun should_show_empty_case_if_there_are_no_todos() {
+    givenNoTodos()
+
+    rule.launchActivity(null)
+
+    todosListScreen {
+      emptyCase {
+        isVisible()
+      }
+    }
+  }
+
+  @Test
+  fun should_show_two_items_if_there_are_two_todos() {
+    givenTwoTodos()
+
+    rule.launchActivity(null)
+
+    todosListScreen {
+      emptyCase { isNotDisplayed() }
+
+      todosList {
+        lastChild<TodosListScreen.TodoItem> {
+          title { hasText("Another todo") }
+        }
+      }
+    }
+  }
+
+  @Test
+  fun should_mark_one_item_with_stroken_title_when_clicking_todo() {
+    givenTwoTodos()
+
+    rule.launchActivity(null)
+
+    todosListScreen {
+      todosList {
+        firstChild<TodosListScreen.TodoItem> {
+          click()
+
+          title {
+            hasText("One todo")
+
+            // Dunno how to test paint flags ¯\_(ツ)_/¯
```

Comment:
You need to create your own matcher, it is very easy.

```
public static Matcher<View> withTextColor(final int style) {
    Checks.checkNotNull(color);
    return new BoundedMatcher<View, TextView>(TextView.class) {
        @Override
        public boolean matchesSafely(TextView warning) {
            int currentStyle = getTypeface().getStyle();
            return style == currentStyle
        }
        @Override
        public void describeTo(Description description) {
            description.appendText("with style: ");
        }
    };
}
```

For example, I haven't tested but it should work something like that.

Reply:
The thing here is that I'm using Kakao for defining a Screen pattern over the Fragment and make these assertions easier. I'll try to use an Espresso matcher inside Kakao. Thank you for the snippet 👍

Reply:
👍 you are welcome